File: SQL Server 2012/Projects/Visual Basic/Chapter 23/ApClrObjects/Triggers/Database_CreateTable_DropTable.vb
Imports System
Imports System.Data
Imports System.Data.SqlClient
Imports System.Data.SqlTypes
Imports Microsoft.SqlServer.Server


Partial Public Class Triggers
    <Microsoft.SqlServer.Server.SqlTrigger(
        Target:="Database",
        Event:="AFTER CREATE_TABLE, DROP_TABLE")> _
    Public Shared Sub Database_CreateTable_DropTable()
        Dim tc As SqlTriggerContext = SqlContext.TriggerContext
        If tc.TriggerAction = TriggerAction.CreateTable Then
            SqlContext.Pipe.Send("A new table has been created.")
        ElseIf tc.TriggerAction = TriggerAction.DropTable Then
            SqlContext.Pipe.Send("A table has been dropped.")
        End If
        Dim eventData As String = tc.EventData.Value
        SqlContext.Pipe.Send(eventData)
    End Sub
End Class

File: SQL Server 2012/Projects/C#/Chapter 23/ApClrObjects/ApClrObjects/Triggers/Database_CreateTable_DropTable.cs
//------------------------------------------------------------------------------
// <copyright file="CSSqlTrigger.cs" company="Microsoft">
//     Copyright (c) Microsoft Corporation.  All rights reserved.
// </copyright>
//------------------------------------------------------------------------------
using System;
using System.Data;
using System.Data.SqlClient;
using Microsoft.SqlServer.Server;

public partial class Triggers
{        
    [Microsoft.SqlServer.Server.SqlTrigger (Target="Database", Event="AFTER CREATE_TABLE, DROP_TABLE")]
    public static void Database_CreateTable_DropTable ()
    {
        SqlTriggerContext tc = SqlContext.TriggerContext;
        if (tc.TriggerAction == TriggerAction.CreateTable)
        {
            SqlContext.Pipe.Send("A new table has been created.");
        }
        else if (tc.TriggerAction == TriggerAction.DropTable)
        {
            SqlContext.Pipe.Send("A table has been dropped.");
        }
        string eventData = tc.EventData.Value;
        SqlContext.Pipe.Send(eventData);
    }
}

